### PR TITLE
Add appId & version to AppMetadata

### PR DESCRIPTION
**Pull request by sean-ciq**
This PR contributes appId & version to the definition of an AppMetadata object as described in #240 

Specifically this PR changes
- The AppMetadata type definition and interface
- Updates documentation

### CODE REVIEW

File: src/api/AppMetadata.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ export interface AppMetadata {
   /** The unique app name that can be used with the open and raiseIntent calls. */
   readonly name: string;
 
+  /** The unique application identifier located within a specific application directory instance.  */
```
Comment (kriswest) [inline src/api/AppMetadata.ts:13]: Add examples of a fully qualified appId to the comments

**Review by kriswest — CHANGES_REQUESTED** [review]
WG feedback - PR/change consider 'harmless' and should pass. Could we add some example appIds to comments

File: src/api/AppMetadata.ts (hunk 1 above)
Comment (rikoe) [reply src/api/AppMetadata.ts:13]: Do we have such examples, or would we just be speculating?

**Review by rikoe — APPROVED** [review]

File: src/api/AppMetadata.ts (hunk 1 above)
Comment (sean-ciq) [reply src/api/AppMetadata.ts]: @kriswest I decided to go with consistency. I added an example to the comment based upon the documentation here https://fdc3.finos.org/docs/1.1/app-directory/discovery

**Pull request merged**